docs/rfc/rfc-026-p2p-bad-peers-checktx.md

```diff
@@ -0,0 +1,375 @@
+  and investigating how [Tendermint treats large Txs](https://github.com/celestiaorg/celestia-core/issues/243)
+- BigChainDb: [Handling proposers who propose bad blocks](https://github.com/bigchaindb/BEPs/issues/84)
+- IBC relayer: Simulate transactions in the mempool and detect whether they could ever have been valid to 
+  [avoid overloading the p2p/mempool system](https://github.com/cosmos/ibc-go/issues/853#issuecomment-1032211020)
+ (ToDo check with Adi what is this about, what are valid reasons this can happen and when should we disconnect)
+  (relevant code [here](https://github.com/cosmos/ibc-go/blob/a0e59b8e7a2e1305b7b168962e20516ca8c98fad/modules/core/ante/ante.go#L23)) 
+- From  Adi: Nodes allow transactions with a wrong `minGas` transaction and gossip them, and other nodes keep rejecting them. (Problem seen with relayers)
+
+### Current state of mempool/p2p interaction
+
+Transactions received from a peer are handled within the [`Receive`](https://github.com/tendermint/tendermint/blob/ff0f98892f24aac11e46aeff2b6d2c0ad816701a/mempool/v0/reactor.go#L158) routine. 
+
+Currently, the mempool triggers a disconnect from a peer in the case of the following errors:
+
+  - [Unknown message type](https://github.com/tendermint/tendermint/blob/ff0f98892f24aac11e46aeff2b6d2c0ad816701a/mempool/v0/reactor.go#L184)
+
+However, disconnecting from a peer is not the same as banning the peer. The p2p layer will close the connecton but 
+the peer can reconnect without any penalty, and if it as a persistent peer, a reconnect will be initiated
+from the node. 
+
+### Current support for peer banning
+
+The p2p layer implements only banning peers for an indefinite amount of time, by marking them
+as bad (calling the `MarkBad` routine implemented by the `Switch`). The peers are reinstated to the list 
+of good peers only if the node requires more peers. They can also not be marked as bad forever (blacklisted). 
+
+The application can blacklist peers via ABCI if the 
+[`filterPeers`](../../spec/abci/abci%2B%2B_app_requirements.md#peer-filtering) 
+config flag is set, by providing a set of peers to ban to Tendermint. 
+
+If the discussion in this RFC deems a different banning mechanism is needed,
+the actual implementation and design of this mechanism will be discussed in a separate RFC.
+This mechanism should be generic, designed within the p2p layer and simply provide an interface
+for reactors to indicate peers to ban and for how long. It should not involve any mempool
+specific design considerations. 
+
+## Discussion
+
+If this feature is to be implemented we need to clearly define the following:
+1. What does banning a peer mean:
+   1. A peer can be simply disconnected from. 
+   2. Peer is disconnected from and banned.
+   3. Conditions for the peer to be banned.
+2. If `CheckTx` signals a peer should be banned, how do we know which peer(s) to ban?
```

Comment:
Not sure I understand this point

Reply:
Yes, initially we discussed about whether the SenderId field is really representative of the set of peers responsible for this message. However, it is so I have rephrased this,